### PR TITLE
vmware_guest: Cloned machine's hostname is no longer updated

### PR DESCRIPTION
**Pull request by ngp-star**
##### SUMMARY
include fix for the issue https://github.com/ansible/ansible/issues/57873
##### ISSUE TYPE
- Bugfix Pull Request
##### COMPONENT NAME
vmware_guest.py
##### ADDITIONAL INFORMATION
code changes done for bug raised under the issue https://github.com/ansible/ansible/issues/57873

### CODE REVIEW

File: lib/ansible/modules/cloud/vmware/vmware_guest.py [hunk 1]
```diff
@@ -2398,6 +2398,9 @@ def deploy_vm(self):
         for nw in self.params['networks']:
             for key in nw:
                 # We don't need customizations for these keys
+                if key == 'type' and nw['type'] == 'dhcp':
```
Comment (pgbidkar) [inline lib/ansible/modules/cloud/vmware/vmware_guest.py:2401]: Add this check inside if key not in ('device_type', 'mac', 'name', 'vlan', 'type', 'start_connected'): condition. Dont make it separate if block

**Review by pgbidkar — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/vmware/vmware_guest.py (hunk 1 above)
Comment (pgbidkar) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py:2401]: Sorry, I misread the logic. Look good to me.

**Comment by pgbidkar** [comment]
Hi @Akasurde , @goneri , Please review this change

Thanks

**Review by goneri — APPROVED** [review]

**Comment by ngp-star** [comment]
@Akasurde @goneri  is there any update?

**Comment by ngp-star** [comment]
@Akasurde 
can you just review this fix?

**Pull request merged**